pkg/kubelet/pod_workers.go

```diff
@@ -87,36 +95,39 @@ func newPodWorkers(runtimeCache kubecontainer.RuntimeCache, syncPodFn syncPodFnT
 		runtimeCache:              runtimeCache,
 		syncPodFn:                 syncPodFn,
 		recorder:                  recorder,
+		workQueue:                 workQueue,
+		resyncInterval:            resyncInterval,
+		backOffInterval:           backOffInterval,
 	}
 }
 
 func (p *podWorkers) managePodLoop(podUpdates <-chan workUpdate) {
 	var minRuntimeCacheTime time.Time
 	for newWork := range podUpdates {
-		func() {
-			defer p.checkForUpdates(newWork.pod.UID, newWork.updateCompleteFn)
+		func() (err error) {
+			defer func() { p.wrapUp(newWork.pod.UID, err) }()
```

Comment:
I think it was scoped right, but agreed that it wasn't easy to read. Fixed based on the suggestion.